Projects/Faker/Faker/Program.cs

```diff
@@ -1,63 +1,72 @@
+
+    // or, if you want JSON:
+    // var userString = JsonSerializer.Serialize(user, new JsonSerializerOptions {WriteIndented = true})
+
+    Console.WriteLine(userString);
+}
+
+User CreateUser()
+{
+    var fullNameInput = Faker.Name.FullName();
+
+    var titles = new[]
     {
-        class User
-        {
-            public string FullName { get; set; }
-            public string FirstName { get; set; }
-            public string LastName { get; set; }
-        }
-        static void Main(string[] args)
+        "Prof",
+        "Dr",
+        "Mr",
+        "Mrs",
+        "Miss",
+        "Ms",
+    };
+
+    var index = 0;
+    foreach (var title in titles.OrderByDescending(x => x.Length))
+    {
+        if (!fullNameInput.StartsWith(title))
         {
-            string json = JsonConvert.SerializeObject(CreateUser());
-            Console.WriteLine(json);
+            continue;
         }
 
-        private static User CreateUser()
+        switch (fullNameInput[title.Length])
         {
-            User user = new User();
-            user.FullName = Faker.Name.FullName();
-            // Check if FullName contains Prof, Dr. Mr. Mrs. or Miss If so remove those.
-            if (user.FullName.Contains("Prof.") || user.FullName.Contains("Dr.") || user.FullName.Contains("Mr.") || user.FullName.Contains("Mrs.") || user.FullName.Contains("Miss") || user.FullName.Contains("Ms"))
-            {
-                user.FirstName = user.FullName.Split(' ')[1];
-                // If FullName string contains more then 2 strings concatinate LastName
-                if (user.FullName.Length > 3)
-                {
-                    String message = String.Format("FullName {0} contains more than 1 LastName {1}.", user.FullName, user.FullName.Split(" ").Length);
-                    int i = (user.FullName.Split(" ").Length);
-                    user.LastName = string.Concat(user.FullName.Split(' ')[2..i]);
-                }
-                else
-                {
-                    user.LastName = user.FullName.Split(' ')[2];
-                }
-                
-            }
-            else
-            {
-                user.FirstName = user.FullName.Split(' ')[0];
-                if (user.FullName.Length > 2)
-                {
-                    int i = (user.FullName.Split(" ").Length);
-                    user.LastName = string.Concat(user.FullName.Split(' ')[1..i]);
-                }
-                else
-                {
-                    user.LastName = user.FullName.Split(' ')[1];
-                }
-                
-            };
-            
-            return user;
+            case ' ':
+                index = title.Length + 1;
+                break;
+            case '.':
+                index = title.Length + 2;
+                break;
+            default:
+                continue;
         }
+
+        break;
     }
 
-    
+    var names = fullNameInput[index..].Split(' ');
```

Comment:
What does [index..] exactly does?

Reply:
A string is a collection of characters. With the range operator, you can select... a range... ;-)
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-8.0/ranges#systemrange
In this case, the starting position of the names (0 when no title, otherwise 2, 3 or 4), unbound: to the end (..).